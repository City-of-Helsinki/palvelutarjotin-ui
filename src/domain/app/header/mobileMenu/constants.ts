export const MOBILE_MENU_ACTIONS = {
  CLOSE_MOBILE_MENU: 'CLOSE_MOBILE_MENU',
  OPEN_MOBILE_MENU: 'OPEN_MOBILE_MENU',
};
