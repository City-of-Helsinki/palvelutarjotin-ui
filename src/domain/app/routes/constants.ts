export enum ROUTES {
  EVENTS_SEARCH = '/search',
  EVENT_DETAILS = '/event/:id',
  CREATE_ENROLMENT = '/event/:id/enrolment/create',
  HOME = '/',
  PRIVACY_POLICY = '/privacy-policy',
  CMS_PAGE = '/cms-page/:slug',
  NEWSLETTER_SUBSCRIBE = '/newsletter/subscribe',
}

export enum PATHNAMES {
  CMS_PAGE = '/cms-page/[...slug]',
}
