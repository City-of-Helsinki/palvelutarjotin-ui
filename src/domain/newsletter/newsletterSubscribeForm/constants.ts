export enum NewsletterGroupId {
  // CULTURAL_EDUCATION = '0f59fb0fe9d0192185ec00aec17290e4',
  ELEMENTARY_SCHOOL = '036765ce8c870d8dc6e2e4805754edc3',
  EARLY_CHILDHOOD_EDUCATION = 'b36d5372abf3d5193102e7fe93a25f47',
  HIGH_AND_SECONDARY_SCHOOLS = '536a600319a708507bb85c0b9ad7b9aa',
}
