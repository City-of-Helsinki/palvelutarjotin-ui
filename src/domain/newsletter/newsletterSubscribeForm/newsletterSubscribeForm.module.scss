@import 'layout';
.newsletterSubscribeForm {
  margin-top: var(--spacing-layout-m);

  .rowWith2Columns {
    display: grid;
    grid-gap: var(--spacing-layout-s);

    @include respond_above(sm) {
      grid-template-columns: 1fr 1fr;
    }
  }
}
