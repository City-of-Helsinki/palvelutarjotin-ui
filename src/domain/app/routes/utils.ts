import { CollectionItemType } from 'react-helsinki-headless-cms';

import { ROUTES } from './constants';

export const getCmsPagePath = (uri?: string | null): string => {
  return ROUTES.CMS_PAGE.replace('/:uri', uri ?? '');
};

export const getCmsArticlePath = (uri?: string | null): string => {
  return ROUTES.CMS_ARTICLE.replace('/:uri', uri ?? '');
};

export const getCollectionItemUrl = (item: CollectionItemType): string => {
  if (!item) {
    return '#';
  }
  if (item.__typename === 'Post') {
    return getCmsArticlePath(item.uri);
  }
  if (item.__typename === 'Page') {
    return getCmsPagePath(item.uri);
  }
  return item?.uri ?? '';
};
