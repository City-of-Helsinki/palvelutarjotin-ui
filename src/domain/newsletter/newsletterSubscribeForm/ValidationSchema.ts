import * as Yup from 'yup';

import { VALIDATION_MESSAGE_KEYS } from '../../app/forms/constants';

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export default function getValidationSchema() {
  return Yup.object().shape({
    groups: Yup.array()
      .required(VALIDATION_MESSAGE_KEYS.STRING_REQUIRED)
      .min(1, VALIDATION_MESSAGE_KEYS.STRING_REQUIRED),
    firstName: Yup.string().required(VALIDATION_MESSAGE_KEYS.STRING_REQUIRED),
    lastName: Yup.string().required(VALIDATION_MESSAGE_KEYS.STRING_REQUIRED),
    email: Yup.string().required(VALIDATION_MESSAGE_KEYS.STRING_REQUIRED),
  });
}
