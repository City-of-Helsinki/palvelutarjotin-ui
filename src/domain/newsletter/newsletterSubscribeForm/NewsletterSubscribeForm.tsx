import { Field, Formik } from 'formik';
import { Button } from 'hds-react';
import React from 'react';
import { useTranslation } from 'react-i18next';

import MultiDropdownField from '../../../common/components/form/fields/MultiDropdownField';
import TextInputField from '../../../common/components/form/fields/TextInputField';
import FocusToFirstError from '../../../common/components/form/FocusToFirstError';
import FormGroup from '../../../common/components/form/FormGroup';
import { translateValue } from '../../../utils/translateUtils';
import { NewsletterGroupId, NewsletterSubscribeFormFields } from './constants';
import styles from './newsletterSubscribeForm.module.scss';
import getValidationSchema from './ValidationSchema';

export const defaultInitialValues: NewsletterSubscribeFormFields = {
  groups: Object.values(NewsletterGroupId),
  firstName: '',
  lastName: '',
  email: '',
};

type Props = {
  initialValues: NewsletterSubscribeFormFields;
  onSubmit: (values: NewsletterSubscribeFormFields) => void;
};

const NewsletterSubscribeForm: React.FC<Props> = ({
  initialValues = defaultInitialValues,
  onSubmit,
}) => {
  const { t } = useTranslation();

  const validationSchema = getValidationSchema();

  const groupsOptions = Object.entries(NewsletterGroupId).map(
    ([translationKey, groupId]) => ({
      label: translateValue(
        'newsletter:newsletterSubscribeForm.newsletterGroups.',
        translationKey,
        t
      ),
      value: groupId,
    })
  );

  return (
    <Formik
      enableReinitialize={true}
      initialValues={initialValues}
      onSubmit={onSubmit}
      validationSchema={validationSchema}
    >
      {({ handleSubmit }) => {
        return (
          <form
            className={styles.newsletterSubscribeForm}
            onSubmit={handleSubmit}
          >
            <FocusToFirstError />
            <FormGroup>
              <Field
                label={t('newsletter:newsletterSubscribeForm.labelGroups')}
                component={MultiDropdownField}
                name="groups"
                options={groupsOptions || []}
                required
              />
            </FormGroup>
            <FormGroup>
              <div className={styles.rowWith2Columns}>
                <Field
                  label={t('newsletter:newsletterSubscribeForm.labelFirstName')}
                  component={TextInputField}
                  name="firstName"
                  required
                />
                <Field
                  label={t('newsletter:newsletterSubscribeForm.labelLastName')}
                  component={TextInputField}
                  name="lastName"
                  required
                />
              </div>
            </FormGroup>
            <FormGroup>
              <div className={styles.rowWith2Columns}>
                <Field
                  label={t('newsletter:newsletterSubscribeForm.labelEmail')}
                  component={TextInputField}
                  name="email"
                  required
                />
              </div>
            </FormGroup>
            <div className={styles.submitButtonWrapper}>
              <Button type="submit">
                {t('newsletter:newsletterSubscribeForm.buttonSubmit')}
              </Button>
            </div>
          </form>
        );
      }}
    </Formik>
  );
};

export default NewsletterSubscribeForm;
