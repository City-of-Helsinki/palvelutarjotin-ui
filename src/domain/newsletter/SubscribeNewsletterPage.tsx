import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { toast } from 'react-toastify';

import TextWithLineBreaks from '../../common/components/textWithLineBreaks/TextWithLineBreaks';
import Container from '../app/layout/Container';
import PageWrapper from '../app/layout/PageWrapper';
import NewsletterSubscribeForm, {
  defaultInitialValues,
  NewsletterSubscribeFormFields,
} from './newsletterSubscribeForm/NewsletterSubscribeForm';
import styles from './subscribeNewsletterPage.module.scss';

const SubscribeNewsletterPage: React.FC = () => {
  const { t } = useTranslation();

  const handleSubmit = async (values: NewsletterSubscribeFormFields) => {
    try {
      toast(t('newsletter:newsletterSubscribeForm.subscribeSuccess'), {
        type: toast.TYPE.SUCCESS,
      });
    } catch (error) {
      toast(t('newsletter:newsletterSubscribeForm.subscribeFailed'), {
        type: toast.TYPE.ERROR,
      });
    }
  };

  return (
    <PageWrapper title={t('newsletter:subscribeNewsletterPage.pageTitle')}>
      <div className={styles.subscribeNewsletter}>
        <Container>
          <h2>{t('newsletter:subscribeNewsletterPage.pageTitle')}</h2>
          <TextWithLineBreaks
            as="p"
            text={t('newsletter:subscribeNewsletterPage.leadText')}
          />
          <NewsletterSubscribeForm
            initialValues={defaultInitialValues}
            onSubmit={handleSubmit}
          />
        </Container>
      </div>
    </PageWrapper>
  );
};

export default SubscribeNewsletterPage;
