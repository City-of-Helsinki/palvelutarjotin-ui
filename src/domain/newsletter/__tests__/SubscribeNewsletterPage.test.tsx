import { http, HttpResponse } from 'msw';
import * as React from 'react';

import { server } from '../../../tests/msw/server';
import { render, userEvent, screen, waitFor } from '../../../utils/testUtils';
import { NewsletterGroupId } from '../newsletterSubscribeForm/constants';
import SubscribeNewsletterPage from '../SubscribeNewsletterPage';

const firstName = 'Test';
const lastName = 'Guy';
const email = 'test.guy@example.com';
const defaultFillValues = {
  firstName,
  lastName,
  email,
};

const serverSuccessfulAddSubscriptionHandlers = Object.values(
  NewsletterGroupId
).map((groupId) =>
  http.post(`/api/newsletter/subscribe/${groupId}`, () => {
    return HttpResponse.json({ email });
  })
);

describe('SubscribeNewsletterPage', () => {
  beforeEach(() => {
    server.use(...serverSuccessfulAddSubscriptionHandlers);
  });

  const fillupForm = async ({
    firstName,
    lastName,
    email,
  } = defaultFillValues) => {
    await userEvent.type(
      screen.getByRole('textbox', {
        name: /etunimi \*/i,
      }),
      firstName
    );

    await userEvent.type(
      screen.getByRole('textbox', {
        name: /sukunimi \*/i,
      }),
      lastName
    );

    await userEvent.type(
      screen.getByRole('textbox', {
        name: /sähköposti \*/i,
      }),
      email
    );
  };

  it('renders correctly', async () => {
    const { container } = render(<SubscribeNewsletterPage />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('handles successful submits properly', async () => {
    render(<SubscribeNewsletterPage />);
    await fillupForm();
    expect(
      screen.queryByRole('heading', {
        name: /tilaus lähetetty/i,
      })
    ).not.toBeInTheDocument();

    await userEvent.click(
      screen.getByRole('button', {
        name: /tilaa uutiskirje/i,
      })
    );

    await waitFor(
      () => {
        expect(screen.getByText(/tilaus lähetetty/i)).toBeInTheDocument();
      },
      { timeout: 10000 }
    );
  });

  it('handles erroneous submits properly', async () => {
    const serverUnsuccessfulAddSubscriptionHandlers = Object.values(
      NewsletterGroupId
    ).map((groupId) =>
      http.post(`/api/newsletter/subscribe/${groupId}`, () => {
        return HttpResponse.error();
      })
    );
    server.use(...serverUnsuccessfulAddSubscriptionHandlers);

    render(<SubscribeNewsletterPage />);

    await fillupForm();

    expect(
      screen.queryByRole('heading', {
        name: /tilaus lähetetty/i,
      })
    ).not.toBeInTheDocument();

    await userEvent.click(
      screen.getByRole('button', {
        name: /tilaa uutiskirje/i,
      })
    );

    await waitFor(() => {
      expect(
        screen.getByText(
          /valitettavasti tilauksen lähettäminen epäonnistui! yritäthän myöhemmin uudelleen\./i
        )
      ).toBeInTheDocument();
    });
  });
});
