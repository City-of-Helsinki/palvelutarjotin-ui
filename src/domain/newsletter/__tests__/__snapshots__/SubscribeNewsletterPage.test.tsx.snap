// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubscribeNewsletterPage renders correctly 1`] = `
<div>
  <main>
    <div
      class="subscribeNewsletter"
    >
      <div
        class="container default"
      >
        <div>
          <h2>
            Tilaa uutiskirje
          </h2>
          <div>
            <p>
              Tilaamalla Kultus-uutiskirjeen hyväksyn, että lähettämäni tiedot tallennetaan Kulttuuri- ja vapaa-aika toimialan Gruppo-tilaajarekisteriin. Uutiskirjeen tilauksen voi perua milloin tahansa. Linkin tietosuojaselosteeseen löydät tämän sivun alalaidasta.
            </p>
            <p>
              Gruppo-tilin hallintaan ja uutiskirjetilaukseen liittyen voit olla yhteydessä Kultuksen ylläpitäjään: 
              <a
                aria-label="Ylläpitäjän sähköposti: kultus@hel.fi"
                href="mailto:kultus@hel.fi"
              >
                kultus@hel.fi
              </a>
              .
            </p>
          </div>
          <form
            class="newsletterSubscribeForm"
          >
            <div
              class="formGroup"
            >
              <div
                class="Select-module_root__Ka5uO Select-module_multiselect__fxewG"
              >
                <label
                  class="FieldLabel-module_label__1zrXK "
                  for="groups-toggle-button"
                  id="groups-label"
                >
                  Tilaan uutiskirjeen
                  <span
                    class="RequiredIndicator-module_indicator__Pt7GX text-input_hds-text-input__required__2K-Bs"
                  >
                    *
                  </span>
                </label>
                <div
                  class="Select-module_wrapper__1WQXs"
                >
                  <div
                    class="SelectedItems-module_selectedItems__2m24K SelectedItems-module_itemsHidden__2ewH3 SelectedItems-module_truncateFirstChild__192oz"
                  >
                    <div
                      class="Tag-module_tag__3bo5S tag_hds-tag__3MKf3 SelectedItems-module_tag__13n3H"
                      id="hds-tag-7"
                      role="button"
                      tabindex="0"
                    >
                      <span
                        aria-labelledby="groups-label hds-tag-7-label"
                        class="Tag-module_label__zCgFr undefined SelectedItems-module_tagLabel__3qEgh"
                        id="hds-tag-7-label"
                      >
                        <span>
                          Alakouluille
                        </span>
                      </span>
                      <button
                        class="Tag-module_deleteButton__1diMR tag_hds-tag__delete-button__33Tgz"
                        id="hds-tag-7-delete-button"
                        tabindex="-1"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          aria-label="cross"
                          class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                          role="img"
                          viewBox="0 0 24 24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M18 7.5L13.5 12L18 16.5L16.5 18L12 13.5L7.5 18L6 16.5L10.5 12L6 7.5L7.5 6L12 10.5L16.5 6L18 7.5Z"
                            fill="currentColor"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </button>
                    </div>
                    <div
                      class="Tag-module_tag__3bo5S tag_hds-tag__3MKf3 SelectedItems-module_tag__13n3H SelectedItems-module_hidden__2hyRx"
                      id="hds-tag-8"
                      role="button"
                      tabindex="-1"
                    >
                      <span
                        aria-labelledby="groups-label hds-tag-8-label"
                        class="Tag-module_label__zCgFr undefined SelectedItems-module_tagLabel__3qEgh"
                        id="hds-tag-8-label"
                      >
                        <span>
                          Varhaiskasvatukselle
                        </span>
                      </span>
                      <button
                        class="Tag-module_deleteButton__1diMR tag_hds-tag__delete-button__33Tgz"
                        id="hds-tag-8-delete-button"
                        tabindex="-1"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          aria-label="cross"
                          class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                          role="img"
                          viewBox="0 0 24 24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M18 7.5L13.5 12L18 16.5L16.5 18L12 13.5L7.5 18L6 16.5L10.5 12L6 7.5L7.5 6L12 10.5L16.5 6L18 7.5Z"
                            fill="currentColor"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </button>
                    </div>
                    <div
                      class="Tag-module_tag__3bo5S tag_hds-tag__3MKf3 SelectedItems-module_tag__13n3H SelectedItems-module_hidden__2hyRx"
                      id="hds-tag-9"
                      role="button"
                      tabindex="-1"
                    >
                      <span
                        aria-labelledby="groups-label hds-tag-9-label"
                        class="Tag-module_label__zCgFr undefined SelectedItems-module_tagLabel__3qEgh"
                        id="hds-tag-9-label"
                      >
                        <span>
                          Yläkouluille ja toisen asteen oppilaitoksille
                        </span>
                      </span>
                      <button
                        class="Tag-module_deleteButton__1diMR tag_hds-tag__delete-button__33Tgz"
                        id="hds-tag-9-delete-button"
                        tabindex="-1"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          aria-label="cross"
                          class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                          role="img"
                          viewBox="0 0 24 24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M18 7.5L13.5 12L18 16.5L16.5 18L12 13.5L7.5 18L6 16.5L10.5 12L6 7.5L7.5 6L12 10.5L16.5 6L18 7.5Z"
                            fill="currentColor"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </button>
                    </div>
                    <span
                      class="SelectedItems-module_hiddenItemsCount__3MAEc"
                    >
                      +
                      2
                    </span>
                  </div>
                  <button
                    class="SelectedItems-module_clearButton__35tnp"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      aria-label="cross-circle"
                      class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                      role="img"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M12 2C17.5228 2 22 6.47715 22 12C22 17.5228 17.5228 22 12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2ZM12 4C7.58172 4 4 7.58172 4 12C4 16.4183 7.58172 20 12 20C16.4183 20 20 16.4183 20 12C20 7.58172 16.4183 4 12 4ZM15 7.5L16.5 9L13.5 12L16.5 15L15 16.5L12 13.5L9 16.5L7.5 15L10.5 12L7.5 9L9 7.5L12 10.5L15 7.5Z"
                        fill="currentColor"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </button>
                  <button
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-labelledby="groups-label groups-toggle-button"
                    aria-owns="groups-menu"
                    class="Select-module_button__1aIsm"
                    id="groups-toggle-button"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      aria-label="angle-down"
                      class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik Select-module_angleIcon__2-9AD"
                      role="img"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M12 13.5L17 8.5L18.5 10L12 16.5L5.5 10L7 8.5L12 13.5Z"
                        fill="currentColor"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </button>
                  <ul
                    aria-labelledby="groups-label"
                    aria-multiselectable="true"
                    aria-required="true"
                    class="Select-module_menu__1H2aU"
                    id="groups-menu"
                    role="listbox"
                    style="max-height: 260px;"
                    tabindex="-1"
                  />
                </div>
              </div>
            </div>
            <div
              class="formGroup"
            >
              <div
                class="rowWith2Columns"
              >
                <div
                  class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
                >
                  <label
                    class="FieldLabel-module_label__1zrXK "
                    for="firstName"
                  >
                    Etunimi
                    <span
                      class="RequiredIndicator-module_indicator__Pt7GX text-input_hds-text-input__required__2K-Bs"
                    >
                      *
                    </span>
                  </label>
                  <div
                    class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
                  >
                    <input
                      class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
                      id="firstName"
                      required=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
                >
                  <label
                    class="FieldLabel-module_label__1zrXK "
                    for="lastName"
                  >
                    Sukunimi
                    <span
                      class="RequiredIndicator-module_indicator__Pt7GX text-input_hds-text-input__required__2K-Bs"
                    >
                      *
                    </span>
                  </label>
                  <div
                    class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
                  >
                    <input
                      class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
                      id="lastName"
                      required=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="formGroup"
            >
              <div
                class="rowWith2Columns"
              >
                <div
                  class="TextInput-module_root__2CMNr text-input_hds-text-input__2LODq"
                >
                  <label
                    class="FieldLabel-module_label__1zrXK "
                    for="email"
                  >
                    Sähköposti
                    <span
                      class="RequiredIndicator-module_indicator__Pt7GX text-input_hds-text-input__required__2K-Bs"
                    >
                      *
                    </span>
                  </label>
                  <div
                    class="TextInput-module_inputWrapper__3Rvel text-input_hds-text-input__input-wrapper__1OqYG"
                  >
                    <input
                      class="TextInput-module_input__1BlHi text-input_hds-text-input__input__GJm5C"
                      id="email"
                      required=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="submitButtonWrapper"
            >
              <button
                class="Button-module_button__1msFE button_hds-button__2A0je Button-module_primary__2LfKB button_hds-button--primary__2NVvO"
                type="submit"
              >
                <span
                  class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
                >
                  Tilaa uutiskirje
                </span>
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </main>
</div>
`;
